### DISCUSSION TITLE
Suspend in gatsby

### DISCUSSION THREAD

**Discussion by vsolanogo** [gatsbyjs/gatsby#27519] [Help]
I have a very complicated form on the very bottom on each page of my site and i want to load all js code related to this form only when user scrolled to it, before that just display ssr HTML, how i do it in gatsby?

**Comment by xaverfleer** [comment]
Have you considered using the [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)?

The API is designed for use cases like yours.

**Comment by vsolanogo** [comment]
> Have you considered using the [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)?
> 
> The API is designed for use cases like yours.

i even use it) but i need to disoplay html with styles but js code load on intersection

**Comment by xaverfleer** [comment]
Hmm.. sounds like an [XY problem](http://xyproblem.info/).

Why do you want to `load all js code [...] only when scrolled to it`? Are you trying to improve performance?

**Comment by wardpeet** [comment]
This phenomenon is called partial hydration where the whole HTML is sent with the network request but javascript for pieces below the fold are postponed. 

We'll look into this in the near future but we haven't started the exploration phase yet.
